### ISSUE TITLE
Mount path based on env var

### ISSUE THREAD

**Issue by devhl-labs** [devcontainers/features#1132]
I am trying to mount ~/.azure into my devcontainer. The issue is that this directory is a symlink to my Windows path. I can get the mount to work if I mount both paths, the problem is that I don't want to commit my Windows path into git because it wont be the same for others. For others, it may not even be a symlink. Is it possible to mount my symlink using an environment variable if the user has it set?

**Comment by samruddhikhandale** [comment]
Yes, you can use an environment variable in your `devcontainer.json` to mount the symlink. Define the variable in an `.env` file and reference it in the `mounts` section.

eg. 

```
{
    "name": "My Dev Container",
    "image": "mcr.microsoft.com/vscode/devcontainers/base:0.202.6",
    "mounts": [
        "source=${AZURE_PATH},target=/root/.azure,type=bind,consistency=cached"
    ],
    "remoteEnv": {
        "AZURE_PATH": "${localEnv:AZURE_PATH}"
    }
}
```

**Comment by devhl-labs** [comment]
This gives me an error `Error response from daemon: ${AZURE_PATH} is not an absolute path.`

```
    "mounts": [
        "type=bind,source=${localEnv:HOME}/.bash_history,target=${localEnv:HOME}/.bash_history",
        "type=bind,source=${localEnv:HOME}/.ssh,target=${localEnv:HOME}/.ssh,readonly",
        "type=bind,source=${localEnv:HOME}/.azure,target=${localEnv:HOME}/.azure",
        "source=${AZURE_PATH},target=/root/.azure,type=bind,consistency=cached"
    ],
    "runArgs": [
        "--env-file",
        ".devcontainer/devcontainer.env"
    ],
    "remoteEnv": {
        "AZURE_PATH": "${localEnv:AZURE_PATH}"
    }
```

**Comment by devhl-labs** [comment]
Mounts and runArgs do not resolve the environment variables, so it seems there is no way to have a dynamic mount. There is a similar issue here https://github.com/microsoft/vscode-remote-release/issues/3972 @samruddhikhandale Is there anything else I can try?

**Comment by devhl-labs** [comment]
The only way I've been able to get dynamic mount paths is by having my .bashrc run the script

bash.rc
```
set -a
source /your/path/.env
set +a
```

.devcontainer/devcontainer.json
```
    "mounts": [
        "type=bind,source=${localEnv:FOO},target=${localEnv:FOO}"
    ],
    "runArgs": [
        "--env-file",
        ".env"
    ],
```

**Comment by developertom01** [comment]
@samruddhikhandale 
I am encountering an issue while attempting to reference a system environment variable within my devcontainer.json configuration. My current setup includes the following configurations:

json
Copy
Edit
"containerEnv": {
    "MY_ENV": "${localEnv:SYS_ENV}"
},
"runArgs": ["--env-file", ".devcontainer/devcontainer.env"]
Problem Description:

Using containerEnv: I have defined MY_ENV to inherit the value of the host's SYS_ENV environment variable using the ${localEnv:SYS_ENV} syntax. However, within the container, MY_ENV appears as an empty string, indicating that the variable is not being correctly passed.

Using runArgs with --env-file: I specified an environment file (.devcontainer/devcontainer.env) containing necessary environment variables. Despite this, the variables defined in this file are not accessible within the container, as they also appear as empty strings.

Steps Taken:

Verified Host Environment Variable: Confirmed that SYS_ENV is set and accessible in the host environment prior to launching Visual Studio Code.

Launched VS Code from Terminal: Started VS Code from a terminal session where SYS_ENV is defined to ensure the editor inherits the correct environment variables.

Checked Environment File Path: Ensured that the path to .devcontainer/devcontainer.env is correct and that the file follows the standard KEY=VALUE format.

Rebuilt the Container: Executed the "Rebuild and Reopen in Container" command to apply the latest configurations.

Despite these efforts, the issue persists. I am seeking guidance on how to properly reference and pass host environment variables into the development container. Any insights or recommendations to resolve this would be greatly appreciated.
